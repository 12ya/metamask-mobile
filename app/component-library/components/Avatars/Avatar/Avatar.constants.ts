/* eslint-disable import/prefer-default-export */
import { AvatarSizes } from './Avatar.types';

export const AVATAR_AVATAR_ICON_TEST_ID = 'avatar-avatar-icon';
export const AVATAR_AVATAR_IMAGE_TEST_ID = 'avatar-avatar-image';
export const AVATAR_AVATAR_INITIAL_TEST_ID = 'avatar-avatar-initial';

export const DEFAULT_AVATAR_SIZE = AvatarSizes.Md;
