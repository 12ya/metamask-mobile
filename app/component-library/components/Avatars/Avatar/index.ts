export { default } from './Avatar';
export { AvatarSizes, AvatarVariants } from './Avatar.types';
