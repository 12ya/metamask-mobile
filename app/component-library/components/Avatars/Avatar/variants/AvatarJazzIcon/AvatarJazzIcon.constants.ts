/* eslint-disable import/prefer-default-export */
import { IJazziconProps } from 'react-native-jazzicon';

const TEST_AVATAR_JAZZICON_ADDRESS =
  '0x10e08af911f2e489480fb2855b24771745d0198b50f5c55891369844a8c57092';

export const TEST_AVATAR_JAZZICON_PROPS: IJazziconProps = {
  address: TEST_AVATAR_JAZZICON_ADDRESS,
};

export const AVATAR_JAZZICON_TEST_ID = 'avatar-jazzicon';
