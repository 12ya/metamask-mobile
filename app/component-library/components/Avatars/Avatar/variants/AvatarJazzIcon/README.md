# AvatarJazzIcon

AvatarJazzIcon is a component that renders an avatar with a JazzIcon image based on an address.

## Props

This component extends [CirclePatternProps](../../../../../patterns/Circles/Circle/Circle.types.ts) from [CirclePattern](../../../../../patterns/Circles/Circle/Circle.tsx) component.

### `size`

Optional enum to select between size variants. 

| <span style="color:gray;font-size:14px">TYPE</span> | <span style="color:gray;font-size:14px">REQUIRED</span> | <span style="color:gray;font-size:14px">DEFAULT</span> |
| :-------------------------------------------------- | :------------------------------------------------------ | :----------------------------------------------------- |
| [AvatarSizes](../../Avatar.types.ts)          | No                                                     | Md                                                     |

### `badgeProps` 

Optional enum for the Badge props.

| <span style="color:gray;font-size:14px">TYPE</span> | <span style="color:gray;font-size:14px">REQUIRED</span> |
| :-------------------------------------------------- | :------------------------------------------------------ |
| [BadgeProps](../../../../Badges/Badge/Badge.types.ts)                                      | No                                                     |

### `badgePosition`

Optional enum to set the position for the Badge.

| <span style="color:gray;font-size:14px">TYPE</span> | <span style="color:gray;font-size:14px">REQUIRED</span> | <span style="color:gray;font-size:14px">DEFAULT</span> |
| :-------------------------------------------------- | :------------------------------------------------------ | :----------------------------------------------------- |
| [AvatarBadgePositions](../../Avatar.types.ts)          | No                                                     | TopRight                                        |

### `jazzIconProps`

Props for the JazzIcon content.

| <span style="color:gray;font-size:14px">TYPE</span> | <span style="color:gray;font-size:14px">REQUIRED</span> |
| :-------------------------------------------------- | :------------------------------------------------------ |
| [IJazziconProps](https://www.npmjs.com/package/react-native-jazzicon)                                              | Yes                                                     |

## Usage

```javascript
const badgeProps = {
  variant: BadgeVariants.Network,
  networkProps: {
    name: SAMPLE_NETWORK_NAME,
    imageProps: {
      source: SAMPLE_NETWORK_IMG_SOURCE,
    }
  }
}

const jazzIconProps = {
  address: SAMPLE_JAZZICON_ADDRESS,
}

<AvatarJazzIcon 
  size={AvatarSizes.Md}
  jazzIconProps={jazzIconProps}
  badgeProps={badgeProps}
  badgePosition={AvatarBadgePositions.BottomRight}/>;
```
