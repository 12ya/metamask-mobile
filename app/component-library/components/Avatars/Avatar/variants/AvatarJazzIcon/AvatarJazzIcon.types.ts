// Third party dependencies.
import { IJazziconProps } from 'react-native-jazzicon';
import { CirclePatternProps } from '../../../../../patterns/Circles/Circle';

// External dependencies.
import { AvatarVariants } from '../../Avatar.types';

export type AvatarJazzIconProps = CirclePatternProps & {
  /**
   * Avatar variants.
   */
  variant?: AvatarVariants.JazzIcon;
  /**
   * Props for the JazzIcon content.
   */
  jazzIconProps: IJazziconProps;
};
