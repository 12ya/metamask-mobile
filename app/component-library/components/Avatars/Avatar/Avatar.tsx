/* eslint-disable react/prop-types */
import React from 'react';

// External dependencies.
import AvatarIcon from './variants/AvatarIcon';
import AvatarImage from './variants/AvatarImage';
import AvatarInitial from './variants/AvatarInitial';
import AvatarJazzIcon from './variants/AvatarJazzIcon';

// Internal dependencies.
import { AvatarProps, AvatarVariants } from './Avatar.types';

const Avatar = (avatarProps: AvatarProps) => {
  switch (avatarProps.variant) {
    case AvatarVariants.Icon:
      return <AvatarIcon {...avatarProps} />;
    case AvatarVariants.Image:
      return <AvatarImage {...avatarProps} />;
    case AvatarVariants.Initial:
      return <AvatarInitial {...avatarProps} />;
    case AvatarVariants.JazzIcon:
      return <AvatarJazzIcon {...avatarProps} />;
    default:
      throw new Error('Invalid Avatar Variant');
  }
};

export default Avatar;
