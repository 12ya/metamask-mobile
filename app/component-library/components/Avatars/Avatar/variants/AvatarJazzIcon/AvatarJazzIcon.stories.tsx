// Third party dependencies.
import React from 'react';
import { boolean, select } from '@storybook/addon-knobs';

// External dependencies.
import { storybookPropsGroupID } from '../../../../../../component-library/constants/storybook.constants';
import { getBadgeStoryProps } from '../../../../Badges/Badge/Badge.stories';
import { AvatarSizes, AvatarBadgePositions } from '../../Avatar.types';
import { DEFAULT_AVATAR_SIZE } from '../../Avatar.constants';

// Internal dependencies.
import AvatarJazzIcon from './AvatarJazzIcon';
import { AvatarJazzIconProps } from './AvatarJazzIcon.types';
import { SAMPLE_JAZZICON_PROPS } from './AvatarJazzIcon.constants';

export const getAvatarJazzIconStoryProps = (): AvatarJazzIconProps => {
  const sizeSelector = select(
    'size',
    AvatarSizes,
    DEFAULT_AVATAR_SIZE,
    storybookPropsGroupID,
  );
  const includesBadge = boolean('includesBadge', false, storybookPropsGroupID);
  let badgePosition, badgeProps;

  if (includesBadge) {
    badgePosition = select(
      'badgePosition',
      AvatarBadgePositions,
      AvatarBadgePositions.TopRight,
      storybookPropsGroupID,
    );
    badgeProps = getBadgeStoryProps();
  }

  return {
    size: sizeSelector,
    badgePosition,
    badgeProps,
    jazzIconProps: SAMPLE_JAZZICON_PROPS,
  };
};
const AvatarJazzIconStory = () => (
  <AvatarJazzIcon {...getAvatarJazzIconStoryProps()} />
);

export default AvatarJazzIconStory;
