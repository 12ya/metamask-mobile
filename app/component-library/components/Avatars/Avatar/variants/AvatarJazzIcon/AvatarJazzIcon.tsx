// Third party dependencies.
import React from 'react';
import Jazzicon from 'react-native-jazzicon';

// External dependencies.
import CirclePattern from '../../../../../patterns/Circles/Circle/Circle';
import { DEFAULT_AVATAR_SIZE } from '../../Avatar.constants';

// Internal dependencies.
import { AvatarJazzIconProps } from './AvatarJazzIcon.types';
import { AVATAR_JAZZICON_TEST_ID } from './AvatarJazzIcon.constants';

const AvatarJazzIcon = ({
  size = DEFAULT_AVATAR_SIZE,
  jazzIconProps,
  ...props
}: AvatarJazzIconProps) => (
  <CirclePattern size={size} {...props} testID={AVATAR_JAZZICON_TEST_ID}>
    <Jazzicon size={Number(size)} {...jazzIconProps} />
  </CirclePattern>
);

export default AvatarJazzIcon;
