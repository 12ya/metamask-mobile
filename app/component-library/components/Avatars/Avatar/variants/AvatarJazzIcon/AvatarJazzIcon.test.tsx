// Third party dependencies.
import React from 'react';
import { shallow } from 'enzyme';

// External dependencies.
import { AvatarSizes, AvatarVariants } from '../../Avatar.types';

// Internal dependencies.
import AvatarJazzIcon from './AvatarJazzIcon';
import {
  TEST_AVATAR_JAZZICON_ADDRESS,
  AVATAR_JAZZICON_TEST_ID,
} from './AvatarJazzIcon.constants';

describe('AvatarJazzIcon - Snapshot', () => {
  it('should render AvatarJazzIcon correctly', () => {
    const wrapper = shallow(
      <AvatarJazzIcon
        variant={AvatarVariants.JazzIcon}
        size={AvatarSizes.Md}
        address={TEST_AVATAR_JAZZICON_ADDRESS}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});

describe('AvatarJazzIcon', () => {
  it('should render AvatarJazzIcon component', () => {
    const wrapper = shallow(
      <AvatarJazzIcon
        variant={AvatarVariants.JazzIcon}
        size={AvatarSizes.Md}
        address={TEST_AVATAR_JAZZICON_ADDRESS}
      />,
    );
    const AvatarJazzIconComponent = wrapper.findWhere(
      (node) => node.prop('testID') === AVATAR_JAZZICON_TEST_ID,
    );
    expect(AvatarJazzIconComponent.exists()).toBe(true);
  });
});
