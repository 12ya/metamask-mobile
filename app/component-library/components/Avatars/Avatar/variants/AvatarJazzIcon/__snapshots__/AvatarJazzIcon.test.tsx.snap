// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarJazzIcon - Snapshot should render AvatarJazzIcon correctly 1`] = `
<CirclePattern
  size="32"
  testID="avatar-jazzicon"
  variant="JazzIcon"
>
  <Jazzicon
    address="0x10e08af911f2e489480fb2855b24771745d0198b50f5c55891369844a8c57092"
    size={32}
  />
</CirclePattern>
`;
